src/Items/ItemFishingRod.h

```diff
@@ -65,195 +65,238 @@ class cItemFishingRodHandler:
 
 	virtual bool OnItemUse(
 		cWorld * a_World, cPlayer * a_Player, cBlockPluginInterface & a_PluginInterface, const cItem & a_Item,
-		int a_BlockX, int a_BlockY, int a_BlockZ, eBlockFace a_BlockFace
+		const Vector3i a_ClickedBlockPos,
+		eBlockFace a_ClickedBlockFace
 	) override
 	{
-		if (a_BlockFace != BLOCK_FACE_NONE)
+		if (a_ClickedBlockFace != BLOCK_FACE_NONE)
 		{
 			return false;
 		}
 
-		auto & Random = GetRandomProvider();
-
 		if (a_Player->IsFishing())
 		{
-			cFloaterCallback FloaterInfo;
-			a_World->DoWithEntityByID(a_Player->GetFloaterID(), FloaterInfo);
-			a_Player->SetIsFishing(false);
-
-			if (FloaterInfo.IsAttached())
+			ReelIn(*a_World, *a_Player);
```

Comment:
The diff here is completely unreadable. Is it just a straight copy-paste refactor of the code?

Reply:
Yes, I just split it into separate functions for readability.